src/snapshot.ts

```diff
@@ -0,0 +1,53 @@
+import { Extension, onDisconnectPayload } from '@hocuspocus/server';
+import * as Y from 'yjs';
+
+export class Snapshot implements Extension {
+  async onDisconnect(data: onDisconnectPayload) {
+    addVersion(data.document);
+  }
+}
+
+/**
+ * Add a snapshot for current doc
+ * ref: https://github.com/yjs/yjs-demos/blob/d8e33e619d6f2da0fae0c6a361286e6901635a9b/prosemirror-versions/prosemirror-versions.js#L26
+ * @param {Y.Doc} doc
+ */
+export const addVersion = (doc: Y.Doc) => {
+  const versions: Y.Array<{
+    date: number;
+    snapshot: Uint8Array;
+  }> = doc.getArray('versions');
```

Comment:
Discussion on Slack: https://g0v-slack-archive.g0v.ronny.tw/index/channel/C2PPMRQGP#ts-1691721806.3289

- We can consider storing `versions` into Elasticsearch doc
    - We may consider passing the same `elasticsearchOpts` to our snapshot extension to avoid coupling with  
`hocuspocus-extension-elasticsearch` -- the snapshot extension can talk to Elasticsearch by itself
- Does not see others putting snapshot into ydoc that is being snapshotted